choreographer/src/main/java/eu/arrowhead/core/choreographer/ChoreographerController.java

```diff
@@ -275,4 +513,121 @@ private void checkPlanRequest(final ChoreographerPlanRequestDTO request, final S
+        final String validatedServiceDefinitionName = request.getServiceDefinitionName();
+        final int validatedVersion = request.getVersion();
+
+        return choreographerDBService.createExecutorResponse(validatedName, validatedAddress, validatedPort, validatedBaseUri, validatedServiceDefinitionName, validatedVersion);
+    }
+
+    private void checkExecutorRequest(final ChoreographerExecutorRequestDTO request, final String origin) {
+        logger.debug("checkExecutorRequest started...");
+
+        if (request == null) {
+            throw new BadPayloadException("Executor is null.", HttpStatus.SC_BAD_REQUEST, origin);
+        }
+
+        if (Utilities.isEmpty(request.getName())) {
+            throw new BadPayloadException(EXECUTOR_NAME_NULL_ERROR_MESSAGE, HttpStatus.SC_BAD_REQUEST, origin);
+        }
+
+        if (Utilities.isEmpty(request.getAddress())) {
+            throw new BadPayloadException(EXECUTOR_ADDRESS_NULL_ERROR_MESSAGE, HttpStatus.SC_BAD_REQUEST, origin);
+        }
+
+        if (request.getPort() == null) {
+            throw new BadPayloadException(EXECUTOR_PORT_NULL_ERROR_MESSAGE, HttpStatus.SC_BAD_REQUEST, origin);
+        }
+
+        if (Utilities.isEmpty(request.getServiceDefinitionName())) {
+            throw new BadPayloadException(EXECUTOR_SD_NULL_ERROR_MESSAGE, HttpStatus.SC_BAD_REQUEST, origin);
+        }
+
+        if (request.getVersion() == null) {
+            throw new BadPayloadException(EXECUTOR_VERSION_NULL_ERROR_MESSAGE, HttpStatus.SC_BAD_REQUEST, origin);
+        }
+
+        final int validatedPort = request.getPort();
+        if (validatedPort < CommonConstants.SYSTEM_PORT_RANGE_MIN || validatedPort > CommonConstants.SYSTEM_PORT_RANGE_MAX) {
+            throw new BadPayloadException("Port must be between " + CommonConstants.SYSTEM_PORT_RANGE_MIN + " and " + CommonConstants.SYSTEM_PORT_RANGE_MAX + ".", HttpStatus.SC_BAD_REQUEST, origin);
+        }
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    private void checkIfPlanHasEveryRequiredProvider (final ChoreographerRunPlanRequestDTO request, final String origin) {
+        ChoreographerPlan plan = choreographerDBService.getPlanById(request.getId());
+
+        Set<String> serviceDefinitionsFromPlan = getServiceDefinitionsFromPlan(plan);
+        Set<String> serviceDefinitionsByProviders = new HashSet<>();
+
+        for (ServiceRegistryResponseDTO dto : choreographerDriver.queryServiceRegistryByServiceDefinitionList(new ArrayList<>(serviceDefinitionsFromPlan)).getData()) {
+            serviceDefinitionsByProviders.add(dto.getServiceDefinition().getServiceDefinition());
+        }
+
+        if (!serviceDefinitionsByProviders.equals(serviceDefinitionsFromPlan)) {
+            throw new BadPayloadException(CHOREOGRAPHER_INSUFFICIENT_PROVIDERS_FOR_PLAN_ERROR_MESSAGE, HttpStatus.SC_BAD_REQUEST, origin);
+        }
```

Comment:
rewrite this. The current implementation does not use version information as it should